### PR TITLE
Add mariadb

### PR DESCRIPTION
**Pull request by headshog**

### CODE REVIEW

File: projects/mariadb/build.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+fi
+
+if [[ $CONFIG = "coverage" ]]
+then
+      export CC=clang
+      export CXX=clang++
+      export CFLAGS="-g -fprofile-instr-generate -fcoverage-mapping"
+      export CXXFLAGS="$CFLAGS"
+      export LINK_FLAGS="$CFLAGS"
+	export SUFFIX="cov"
```
Comment (koltiradw) [inline projects/mariadb/build.sh:58]: extra space

File: projects/mariadb/fuzz_json.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/* Copyright 2023 Google LLC
```
Comment (apach301) [inline projects/mariadb/fuzz_json.c:1]: Better format license with `//` (like in tinyxml2/main.c)

File: projects/mariadb/fuzz_json.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  json_locate_key(fuzz_str, fuzz_str + size, fuzz_key, &key_start, &key_end,
+                  &comma_pos);
+
+  free(fuzz_str);
+  free(fuzz_key);
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  fuzz_get_object_get(data, size);
+  fuzz_json_locate_key(data, size);
```
Comment (apach301) [inline projects/mariadb/fuzz_json.c:75]: Looks like it should be split in two separate fuzz targets

File: projects/mariadb/build.sh [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+INCLUDE_DIRS="-I/server/wsrep-lib/include -I/server/wsrep-lib/wsrep-API/v26 -I/server/build/include -I/server/include/providers -I/server/include -I/server/sql -I/server/regex -I/server/unittest/mytap"
+$CC $CFLAGS $INCLUDE_DIRS -c /fuzz_json.c -o ./fuzz_json.o
+
+# Link with CXX to support centipede
+$CXX $LINK_FLAGS fuzz_json.o -o /fuzz_json_$SUFFIX \
+	-Wl,--start-group ./unittest/mytap/libmytap.a ./strings/libstrings.a \
+	./dbug/libdbug.a ./mysys/libmysys.a -Wl,--end-group -lz -ldl -lpthread $MAIN_OBJ
+
+# Get corpus.
+if [[ $CONFIG = "coverage" ]]
```
Comment (apach301) [inline projects/mariadb/build.sh:89]: Corpus collection is usually placed at the end of the Dockerfile

File: projects/mariadb/build.sh [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+export MAIN_OBJ=""
+
+if [[ $CONFIG = "libfuzzer" ]]
+then
+      export CC="clang"
```
Comment (apach301) [inline projects/mariadb/build.sh:23]: format: 4 spaces as tabulation

File: projects/mariadb/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+MariaDB was designed as a drop-in replacement of MySQL(R) with more features, new storage engines, fewer bugs, and better performance.
+
+## Build Docker
+
+    $ sudo docker build -t oss-sydr-fuzz-mariadb .
+
+## Run Hybrid Fuzzing
+
+Unzip Sydr (`sydr.zip`) in `projects/ffmpeg` directory:
```
Comment (apach301) [inline projects/mariadb/README.md:11]: ffmpeg -> mariadb

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/mariadb/build.sh [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+export MAIN_OBJ=""
+
+if [[ $CONFIG = "libfuzzer" ]]
+then
+    export CC="clang"
+    export CXX="clang++"
+    export CFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+    export CXXFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+    export LINK_FLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+	export SUFFIX="fuzz"
```
Comment (apach301) [inline projects/mariadb/build.sh:28]: Replace tabs with 4 spaces

**Review by apach301 — CHANGES_REQUESTED** [review]

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/mariadb/build.sh [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+fi
+
+if [[ $CONFIG = "sydr" ]]
+then
+    export CC=clang
+    export CXX=clang++
+    export CFLAGS="-g"
+    export CXXFLAGS="$CFLAGS"
+    export LINK_FLAGS="$CFLAGS"
+	export SUFFIX="sydr"
```
Comment (apach301) [inline projects/mariadb/build.sh:48]: indent

File: projects/mariadb/build.sh [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+fi
+
+if [[ $CONFIG = "afl" ]]
+then
+    export CC="afl-clang-fast"
+    export CXX="afl-clang-fast++"
+    export CFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+    export CXXFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+    export LINK_FLAGS="-g -fsanitize=fuzzer,address,integer,bounds,null,undefined,float-divide-by-zero"
+	export SUFFIX="afl"
```
Comment (apach301) [inline projects/mariadb/build.sh:38]: indent

File: projects/mariadb/build.sh [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+cd server
+rm -rf build
+mkdir build
+cd build
+cmake ../ -DDISABLE_SHARED=ON -LH
+make clean
+
+# Ensure we do static linking
+sed -i 's/libmariadb SHARED/libmariadb STATIC/g' ../libmariadb/libmariadb/CMakeLists.txt
```
[the remark this answers is not in the document]
Comment (apach301) [reply projects/mariadb/build.sh:66]: It's not building that way...
But it works if move that line between cmake and make commands

File: projects/mariadb/build.sh [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+
+INCLUDE_DIRS="-I/server/wsrep-lib/include -I/server/wsrep-lib/wsrep-API/v26 -I/server/build/include -I/server/include/providers -I/server/include -I/server/sql -I/server/regex -I/server/unittest/mytap"
+$CC $CFLAGS $INCLUDE_DIRS -c /fuzz_json.c -o ./fuzz_json.o
+
+INCLUDE_DIRS="-I/server/wsrep-lib/include -I/server/wsrep-lib/wsrep-API/v26 -I/server/build/include -I/server/include/providers -I/server/include -I/server/sql -I/server/regex -I/server/unittest/mytap"
+$CC $CFLAGS $INCLUDE_DIRS -c /fuzz_get.c -o ./fuzz_get.o
+
+# Link with CXX to support centipede
+$CXX $LINK_FLAGS fuzz_json.o -o /fuzz_json_$SUFFIX \
+	-Wl,--start-group ./unittest/mytap/libmytap.a ./strings/libstrings.a \
```
Comment (apach301) [inline projects/mariadb/build.sh:85]: Indentation here and below

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/mariadb/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+WORKDIR /
+
+COPY build.sh *.c /
+
+RUN CONFIG="libfuzzer" /build.sh
+RUN CONFIG="afl" /build.sh
+RUN CONFIG="sydr" /build.sh
+RUN CONFIG="coverage" /build.sh
+
+RUN rm -rf /corpus && mkdir /corpus && /server/mysql-test/suite && \
```
Comment (apach301) [inline projects/mariadb/Dockerfile:35]: cd /server/mysql-test/suite ?

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/mariadb/build.sh (hunk 7 above)
Comment (headshog) [reply projects/mariadb/build.sh]: done

File: projects/mariadb/build.sh (hunk 1 above)
Comment (headshog) [reply projects/mariadb/build.sh]: removed

File: projects/mariadb/build.sh [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+cd server
+rm -rf build
+mkdir build
+cd build
+cmake ../ -DDISABLE_SHARED=ON -LH
+make clean
+
+# Ensure we do static linking
+sed -i 's/libmariadb SHARED/libmariadb STATIC/g' ../libmariadb/libmariadb/CMakeLists.txt
+make
```
[the remark this answers is not in the document]
Comment (headshog) [reply projects/mariadb/build.sh]: done

File: projects/mariadb/build.sh [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    export LINK_FLAGS="$CFLAGS"
+    export SUFFIX="cov"
+fi
+
+cd server
+rm -rf build
+mkdir build
+cd build
+cmake ../ -DDISABLE_SHARED=ON -LH
+make clean
```
[the remark this answers is not in the document]
Comment (headshog) [reply projects/mariadb/build.sh]: removed

File: projects/mariadb/build.sh (hunk 10 above)
[the remark this answers is not in the document]
Comment (headshog) [reply projects/mariadb/build.sh:67]: it must be here

File: projects/mariadb/build.sh [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+# Ensure we do static linking
+sed -i 's/libmariadb SHARED/libmariadb STATIC/g' ../libmariadb/libmariadb/CMakeLists.txt
+make
+rm CMakeCache.txt
```
[the remark this answers is not in the document]
Comment (headshog) [reply projects/mariadb/build.sh]: removed

File: projects/mariadb/build.sh (hunk 8 above)
Comment (headshog) [reply projects/mariadb/build.sh]: done

File: projects/mariadb/build.sh (hunk 9 above)
Comment (headshog) [reply projects/mariadb/build.sh]: done

File: projects/mariadb/build.sh (hunk 11 above)
Comment (headshog) [reply projects/mariadb/build.sh]: done

File: projects/mariadb/Dockerfile (hunk 12 above)
Comment (headshog) [reply projects/mariadb/Dockerfile]: fixed

File: projects/mariadb/build.sh (hunk 9 above)
Comment (headshog) [reply projects/mariadb/build.sh]: done

**Review by apach301 — APPROVED** [review]

File: projects/mariadb/fuzz_get.c [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  if (size < FUZZ_KEY_SIZE) {
+    return -1;
+  }
+  char *fuzz_key = malloc(FUZZ_KEY_SIZE + 1);
+  memcpy(fuzz_key, data, FUZZ_KEY_SIZE);
+  fuzz_key[FUZZ_KEY_SIZE] = '\0';
+
+  data += FUZZ_KEY_SIZE;
```
Comment (kobrineli) [inline projects/mariadb/fuzz_get.c:36]: add valid corpus files according to data format (key + json)

**Review by kobrineli — CHANGES_REQUESTED** [review]

File: projects/mariadb/fuzz_get.c (hunk 16 above)
Comment (headshog) [reply projects/mariadb/fuzz_get.c:36]: added

**Review by kobrineli — CHANGES_REQUESTED** [review]
remove `corpus.zip`

**Review by kobrineli — APPROVED** [review]

**Pull request merged**